crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -99,18 +116,93 @@ impl MultiFarming {
                             plot.clone(),
                             plot_commitments.clone(),
                             client.clone(),
-                            identity,
+                            identity.clone(),
                             reward_address,
                         )
                     });
 
-                    Ok::<_, anyhow::Error>((plot, subspace_codec, plot_commitments, farming))
+                    Ok::<_, anyhow::Error>((
+                        identity,
+                        plot,
+                        subspace_codec,
+                        plot_commitments,
+                        farming,
+                    ))
                 })
             })
-            .collect::<Vec<_>>();
+            .collect::<FuturesOrdered<_>>()
+            .enumerate();
+
+        while let Some((i, result)) = results.next().await {
+            let (identity, plot, subspace_codec, plot_commitments, farming) = result.unwrap()?;
+
+            let mut listen_on = listen_on.clone();
+
+            for multiaddr in &mut listen_on {
+                if let Some(Protocol::Tcp(starting_port)) = multiaddr.pop() {
+                    multiaddr.push(Protocol::Tcp(starting_port + i as u16));
+                    multiaddr.push(Protocol::P2p(
+                        PeerId::from_public_key(&identity::PublicKey::Sr25519(
+                            (*identity.public_key()).into(),
+                        ))
+                        .into(),
+                    ));
```

Comment:
Why do you push this here? It is not necessary for listening unless I'm missing something.